### PR TITLE
Remove jQuery as a dependency

### PR DESCRIPTION
**Pull request by plessbd**
## Description

This removes the jquery library as a dependency from the public facing XDMoD
It currently leaves it in the internal dashboard as it is required by the supreem module

## Motivation and Context

Most of what jQuery provides is already provided by extJs.

the things were were using jQuery for

### extend:
This has been added as XDMoD.utils.extend

### base64:

Could be removed if we did not support IE9
https://caniuse.com/#feat=atob-btoa
https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/atob#Browser_compatibility
https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/btoa#Browser_compatibility

However, since I could not verify that there were still not people using IE9 it was replaced with a small library that did not depend on jquery.

https://github.com/davidchambers/Base64.js

### trim:
Converted as we do not support ie8
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim#Browser_compatibility

## Tests performed

Loaded different pieces of the UI and breakpointed to verify that what was produced is still the same with the new code

Installed Win7 ie9 vm from Microsoft.  Verified that window.atob and window.btoa existed and produced the expected output.

## Types of changes

- [x] Code Cleanup

## Checklist:

- [ ] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by plessbd** [comment]
## further motivation and context
jQuery 1.x is unsupported and to quote the maintainers:

We don't give any guarantees for jQuery 1.x/2.x - they're officially unsupported - but if a serious vulnerability was detected (& sent to security@jquery.com) we **might** reconsider a patch.

https://github.com/jquery/jquery.com/issues/162#issuecomment-298656430

File: composer.json [hunk 1]
```diff
@@ -162,15 +162,15 @@
...
+                "name": "davidchambers/Base64.js-file",
                 "type": "vanilla-plugin",
                 "version": "1.0",
-                "license": "MIT",
-                "homepage": "https://github.com/carlo/jquery-base64",
+                "license": "Apache 2.0 and WTFPL",
+                "homepage": "https://github.com/davidchambers/Base64.js",
                 "dist": {
-                    "url": "https://github.com/carlo/jquery-base64/raw/master/jquery.base64.js",
+                    "url": "https://raw.githubusercontent.com/davidchambers/Base64.js/master/base64.min.js",
```
Comment (chakrabortyr) [inline composer.json]: Not sure I like the concept of adding this. Can you expand your utils namespace?

File: html/gui/js/CCR.js [hunk 2]
```diff
@@ -404,6 +404,48 @@ XDMoD.validator.email = function (email) {
 // A namespace containing commonly-used utility functions.
 
 /**
+ * The jQuery $.extend() API merges the content of subsequent objects into the
+ * first one, overriding it’s original values. The method provided below returns
+ * a new object instead, preserving all of the original objects and their
+ * properties. Supported back to IE6.
+ * https://gomakethings.com/vanilla-javascript-version-of-jquery-extend/
+ */
+XDMoD.utils.extend = function () {
```
Comment (chakrabortyr) [inline html/gui/js/CCR.js]: This file is already huge, can we refactor this to be somewhere else?

File: composer.json (hunk 1 above)
Comment (plessbd) [inline composer.json]: It does base64 encoding and decoding.  If you want to view the unminified source you can view it at https://raw.githubusercontent.com/davidchambers/Base64.js/master/base64.js

It is just a replacement for the jquery.base64.js that was there.

File: html/gui/js/CCR.js (hunk 2 above)
Comment (plessbd) [inline html/gui/js/CCR.js]: It could be refactored to have the different XDMoD.* in their own files, however I would consider that a different change.

**Comment by plessbd** [comment]
technically this does remove it completely from core xdmod.
A pull request to xdmod-supremm would be needed to remove the requirement for it in there.
If it is decided that this will be accepted I will update xdmod-supremm with the updated code changes and update this to remove jquery resource.

**Comment by jpwhite4** [comment]
I'm still not condoning this change, but I do note that, **if** you remove jQuery you should also remove it from the eslintrc definitions (and possibly the gitignore).

**Comment by plessbd** [comment]
to many people want to keep an unneeded library...

**Pull request closed without merging**